x/stakeibc/keeper/ibc.go

```diff
@@ -17,11 +17,10 @@ import (
 )
 
 func (k Keeper) OnChanOpenAck(ctx sdk.Context, portID, channelID string) error {
-	// Lookup the ICA address and chainId from the port and connection
-	controllerConnectionId, found := k.GetConnectionIdFromICAPortId(ctx, portID)
-	if !found {
-		k.Logger(ctx).Info(fmt.Sprintf("portId %s has no associated ICA account", portID))
-		return nil
+	// Lookup connection ID, counterparty chain ID, and ICA address from the channel ID
+	controllerConnectionId, _, err := k.IBCKeeper.ChannelKeeper.GetChannelConnection(ctx, portID, channelID)
```

Comment:
> The OnChanOpenAck callback uses the above erroneous function which prevents registering ICAs on a new connection (since it will always match to the old expired connection)

Was the host zone connection updated?

Reply:
yeah the specific use casa here was for dydx on testnet. The LC expired so we had to make new clients/connections

But even when the new connection ID was passed to register host zone, the lookup in OnChanOpenAck was grabbing the old ICAs on the old channels